src/nogpu/handle.cpp

```diff
@@ -0,0 +1,232 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/config.h>
+#include <miopen/handle.hpp>
+#include <miopen/binary_cache.hpp>
+#include <miopen/device_name.hpp>
+#include <miopen/errors.hpp>
+#include <miopen/gemm_geometry.hpp>
+#include <miopen/handle_lock.hpp>
+#include <miopen/invoker.hpp>
+#include <miopen/kernel_cache.hpp>
+#include <miopen/logger.hpp>
+#include <miopen/timer.hpp>
+
+#if !MIOPEN_ENABLE_SQLITE_KERN_CACHE
+#include <miopen/write_file.hpp>
+#endif
+
+#include <boost/filesystem.hpp>
+#include <miopen/handle_lock.hpp>
+#include <miopen/load_file.hpp>
+#include <miopen/gemm_geometry.hpp>
+
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
+#include <algorithm>
+#include <cassert>
+#include <chrono>
+#include <thread>
+#include <miopen/nogpu/handle_impl.hpp>
+namespace miopen {
+
+Handle::Handle(miopenAcceleratorQueue_t /* stream */) : impl(new HandleImpl())
+{
+#if MIOPEN_USE_ROCBLAS
+    rhandle_ = CreateRocblasHandle();
+#endif
+    MIOPEN_LOG_NQI(*this);
+}
+
+Handle::Handle() : impl(new HandleImpl())
+{
+#if MIOPEN_USE_ROCBLAS
+    rhandle_ = CreateRocblasHandle();
+#endif
+    MIOPEN_LOG_NQI(*this);
+}
+
+Handle::~Handle() {}
+
+void Handle::SetStream(miopenAcceleratorQueue_t /* streamID */) const {}
+
+miopenAcceleratorQueue_t Handle::GetStream() const { return {}; }
+
+void Handle::SetAllocator(miopenAllocatorFunction /* allocator */,
+                          miopenDeallocatorFunction /* deallocator */,
+                          void* /* allocatorContext */) const
+{
+}
+
+void Handle::EnableProfiling(bool enable) const { this->impl->enable_profiling = enable; }
+
+float Handle::GetKernelTime() const { return this->impl->profiling_result; }
+
+Allocator::ManageDataPtr Handle::Create(std::size_t sz) const
+{
+    MIOPEN_HANDLE_LOCK
+    this->Finish();
+    return this->impl->allocator(sz);
+}
+
+Allocator::ManageDataPtr&
+Handle::WriteTo(const void* /* data */, Allocator::ManageDataPtr& ddata, std::size_t /* sz */) const
+{
+    MIOPEN_HANDLE_LOCK
+    this->Finish();
+    return ddata;
+}
+
+void Handle::ReadTo(void* /* data */,
+                    const Allocator::ManageDataPtr& /* ddata */,
+                    std::size_t /* sz */) const
+{
+    MIOPEN_HANDLE_LOCK
```

Comment:
This handle lock is actually for the GPU, so I dont think its needed here.

Reply:
Fixed

Reply:
Removed